Dockerfile

```diff
@@ -1,18 +1,19 @@
-ARG BUILD_IMAGE=public.ecr.aws/eksctl/eksctl-build:26e3c36557da8ae2db5d995cea673e05cc60cfce
-FROM $BUILD_IMAGE as build
+# syntax=docker/dockerfile:1
+FROM public.ecr.aws/docker/library/golang:1.24.0 AS builder
 
 WORKDIR /src
+COPY . .
 
-COPY . /src
+RUN --mount=type=cache,target=/go/pkg/mod <<EOT
+    go mod download
+EOT
 
-RUN make test
-RUN make build \
-    && cp ./eksctl /out/usr/local/bin/eksctl
-RUN make build-integration-test \
-    && mkdir -p /out/usr/local/share/eksctl \
-    && cp -r integration/data/*.yaml integration/scripts /out/usr/local/share/eksctl \
-    && cp ./eksctl-integration-test /out/usr/local/bin/eksctl-integration-test
+RUN <<EOT
+    make install-tools
+    make build
+    chown 65532 eksctl
+EOT
 
-FROM scratch
-COPY --from=build /out /
-ENTRYPOINT ["eksctl"]
+FROM public.ecr.aws/eks-distro/kubernetes/go-runner:v0.16.4-eks-1-32-6 AS go-runner
```

Comment:
Why switch to using a base image instead of scratch?

Reply:
scratch images are not ideal - typically you want something with at least ca-certificates, timezone data, etc.

however, as noted above, this was just following the convention of other EKS projects 
- [Ref](https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/c486a6fdbe1e1982f0852fcd7360c91de6367d4f/Dockerfile#L26)
- [Ref](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/dba6d7090f33e5ccd1452214691e0d1433d7881a/Dockerfile#L31-L44)